File: ai-studio-service/src/main/java/com/aistudio/service/entity/Article.java
package com.aistudio.service.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.time.LocalDateTime;
import java.util.List;

@Data
@TableName("article")
public class Article {

    @TableId(type = IdType.AUTO)
    private Long id;

    private String title;
    private String content;
    private String summary;
    private String coverImage;
    private Long authorId;
    private String authorName;
    private Long folderId;
    private Integer viewsCount;
    private Integer likesCount;
    private Integer commentsCount;
    private Integer favoriteCount;
    private Integer followersCount;
    /** 状态: 0-草稿 1-已发布 2-已下架 */
    private Integer status;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
    /** 定时发布时间 */
    private LocalDateTime publishedAt;
    /** 逻辑删除: 0-正常 1-已删除 */
    private Integer isDeleted;

    @TableField(exist = false)
    private List<Long> tagIds;
}

File: apps/ai-studio-portal-web/shared/types/article.ts
/**
 * 文章类型
 */
export interface Article {
  id: string
  title: string
  content: string
  summary?: string
  coverImage?: string
  authorId: string
  authorName: string
  authorAvatar?: string
  tags: string[]
  categoryId?: string
  categoryName?: string
  viewCount: number
  viewsCount?: number
  likeCount: number
  likesCount?: number
  commentCount: number
  commentsCount?: number
  favoriteCount: number
  status: 'draft' | 'published' | 'deleted'
  createTime: string
  createdAt?: string
  updateTime: string
  updatedAt?: string
  publishTime?: string
  publishedAt?: string
}

/**
 * 文章列表查询参数
 */
export interface ArticleListQuery {
  page?: number
  pageSize?: number
  type?: 'article' | 'question'
  keyword?: string
  tagId?: string
  categoryId?: string
  sort?: 'latest' | 'hot' | 'recommend'
  userId?: string
}

/**
 * 文章列表响应
 */
export interface ArticleListResponse {
  records: Article[]
  total: number
}

/**
 * 文章创建/更新参数
 */
export interface ArticleFormData {
  id?: string
  title: string
  content: string
  summary?: string
  coverImage?: string
  tags: string[]
  categoryId?: string
}
